index.d.ts

```diff
@@ -1,288 +1,600 @@
+// TypeScript Version: 3.0
+
 declare module 'react-table' {
-  import { ReactNode, useState } from 'react'
+  import { ReactNode, useState, ComponentType, MouseEvent } from 'react'
 
-  type StringKey<D> = Extract<keyof D, string>
-  type IdType<D> = StringKey<D> | string
+  /**
+   * The empty definitions of below provides a base definition for the parts used by useTable, that can then be extended in the users code.
+   *
+   * @example
+   *  export interface TableOptions<D extends object = {}}>
+   *    extends
+   *      UseExpandedOptions<D>,
+   *      UseFiltersOptions<D> {}
+   */
+  export interface TableOptions<D extends object> extends UseTableOptions<D> {}
```

Comment:
is `export` needed here?
Aren't all items in a module namespace automatically exported?

Reply:
I don't believe that the items are automatically exported. I can't find any documentation referring to it  and their module example explicitly is exporting items from within the `declare module` block.

Reply:
that may be, I re-targeted #1545 at this branch, it flagged

```
ERROR: 3:1     no-single-declare-module         File has only 1 module declaration — write it as an external module. See: https://github.com/Microsoft/dtslint/blob/master/docs/no-single-declare-module.md
ERROR: 15:3    strict-export-declare-modifiers  'export' keyword is redundant here because all declarations in this module are exported automatically. If you have a good reason to export some declarations and not others, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
```

as priority items to look at.

Reply:
Cleaned up the majority of the linting things. I ignored some of the generic stuff.